fe/src/main/java/org/apache/doris/analysis/ExpressionFunctions.java

```diff
@@ -162,13 +160,52 @@ public FEFunctionSignature getSignature() {
             return signature;
         }
 
+        // Now we only support functions which contain only one variable-length arg or fixed-length args 
+        // which does't contain any arrays.
         public LiteralExpr invoke(List<Expr> args) throws AnalysisException {
+            boolean isArray = false;
+            for (Class<?> argType : method.getParameterTypes()) {
+                if (argType.isArray()) {
+                    Preconditions.checkArgument(method.getParameterTypes().length == 1);
+                    isArray = true;
+                }
+            }
             try {
-                return (LiteralExpr) method.invoke(null, args.toArray());
+                if (!isArray) {
+                    return invokeFixedVariableLengthArgsMethod(args);
+                } else {
+                    return invokeVariableLengthArgsMethod(args);
+                }
             } catch (InvocationTargetException | IllegalAccessException | IllegalArgumentException e) {
                 throw new AnalysisException(e.getLocalizedMessage(), e);
             }
         }
+
+        public LiteralExpr invokeFixedVariableLengthArgsMethod(List<Expr> args) throws InvocationTargetException, IllegalAccessException {
+            return (LiteralExpr) method.invoke(null, args.toArray());
+        }
+
+        public LiteralExpr invokeVariableLengthArgsMethod(List<Expr> args) throws InvocationTargetException, IllegalAccessException {
```

Comment:
I'm afraid that your change can't work with variable function such as `func(int, string...)`

Reply:
i will improve it.